src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js

```diff
@@ -37,10 +38,32 @@ export default ajaxCaller(class LaunchAnalysisModal extends Component {
     }
   }
 
+  async componentDidMount() {
+    const { workspaceId: { namespace, name }, entitySelectionModel: { type, selectedEntities }, processSingle, config: { rootEntityType }, ajax: { Workspaces } } = this.props
+
+    if (!processSingle) {
+      const entities = await Promise.resolve(Utils.switchCase(type,
+        [EntitySelectionType.processAll, async () => _.map('name', await Workspaces.workspace(namespace, name).entitiesOfType(rootEntityType))],
+        [EntitySelectionType.chooseRows, () => _.keys(selectedEntities)],
+        [EntitySelectionType.processMergedSet, () => _.flow(
+          _.flatMap(`attributes.${rootEntityType}s.items`),
+          _.map('entityName')
+        )(selectedEntities)],
+        [Utils.DEFAULT, () => selectedEntities]
+      ))
+
+      this.setState({ entities })
```

Comment:
TOL: We already have the `entityMetadata` loaded on the page, which includes counts. If we passed that in, we might be able to generate the info we need without any additional network calls, and leave the launch code paths as-is.